### PR TITLE
Wallet sort validator list

### PR DESCRIPTION
**Pull request by Jibz1**
https://user-images.githubusercontent.com/8676844/213897773-84bf14d3-dc00-41c9-8b53-a430a437cce8.mov

https://user-images.githubusercontent.com/8676844/214136581-c5bb37d5-7f20-4b9e-9e97-7b690f5d8c5f.mov

### CODE REVIEW

File: apps/wallet/src/ui/app/staking/validators/SelectValidatorCard.tsx
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/staking/validators/SelectValidatorCard.tsx:68]: Rather than having two operations for asc vs desc sort order, you can just flip the result, which will keep sort order still determined by one operation.

File: apps/wallet/src/ui/app/staking/validators/SelectValidatorCard.tsx
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/staking/validators/SelectValidatorCard.tsx:71]: Probably easier to use `String.prototype.localeCompare`:
`a[sortKey].localeCompare(b[sortKey])`

File: apps/wallet/src/ui/app/staking/validators/SelectValidatorCard.tsx
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/staking/validators/SelectValidatorCard.tsx:59]: Why do we always flip the asc / desc order when changing keys? Seems like we should just change the sort key if it changes, and only change asc / desc when re-clicking a selected key.

File: apps/wallet/src/ui/app/staking/validators/SelectValidatorCard.tsx
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/staking/validators/SelectValidatorCard.tsx:101]: Curious why we need negative y margin here. Not always an issue but can point to larger layout issues.

File: apps/wallet/src/ui/app/staking/validators/SelectValidatorCard.tsx
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/staking/validators/SelectValidatorCard.tsx:145]: Not a blocking issue but we should try to start using `@mysten/icons`.

**Review by Jordan-Mysten — APPROVED** [review]
Approving to unblock. Some small feedback, some of which should probably be addressed before merging.

File: apps/wallet/src/ui/app/staking/validators/SelectValidatorCard.tsx
Comment (Jibz1) [reply apps/wallet/src/ui/app/staking/validators/SelectValidatorCard.tsx:101]: There is default padding top and bottom of the parent container. This content should be at full height.

**Review by mystie711 — APPROVED** [review]

**Comment by mystie711** [comment]
@Jibz1 just noticing that the space between the two sentences is a bit tighter in code than in Figma.

image

image

**Comment by Jibz1** [comment]
@mystie711 see updated video

**Review by mystie711 — APPROVED** [review]
looking good.

**Pull request merged**
